DESIGN.md

```diff
@@ -325,18 +325,158 @@ In the addition of the previous rules, an operator for `Flowable`:
 
 ### Creation
 
-Creation of the various types should be exposed through factory methods that provide safe construction.
+Unlike RxJava 1.x, 2.x base classes are to be abstract, stateless and generally no longer wrap an `OnSubscribe` callback - this saves allocation in assembly time without limiting the expressiveness. Operator methods and standard factories still live as final on the base classes.
+
+Instead of the indirection of an `OnSubscribe` and `lift`, operators are to be implemented by extending the base classes. For example, the `map`
```

Comment:
If I am understanding your terminology correctly, you are saying that a "cross base-type" operator is an operator which functions for both an `Observable` and a `Flowable` for instance. Am I correct? I'm suggesting that an operator which knows how to negotiate backpressure from a `Subscriber` should be capable of supporting any kind of `Consumable` that functions using `Subscriber`. This means that there can be many types which implement `Consumable<Observer<T>>` and any operator that knows how to work with an `Observer` can be reused.